docs/developer/solutions/introduction.rst

```diff
@@ -0,0 +1,115 @@
+Introduction
+============
+
+With a focus on having minimal human actions required, both in its deployment
+and operation, MetalK8s also intends to ease deployment and operation of
+complex applications, named *Solutions*, on its cluster.
+
+This document defines what a *Solution* refers to, the responsibilities of each
+party in this integration, and will link to relevant documentation pages for
+detailed information.
+
+What is a *Solution*?
+---------------------
+
+We use the term *Solution* to describe a packaged Kubernetes application,
+archived as an ISO disk image, containing:
+
+- A set of OCI images to inject in MetalK8s image registry
+- An `Operator`_, to deploy on the cluster
+- Optionally, a UI for managing and monitoring the application, represented by
+  a standard Kubernetes ``Deployment``
+
+.. _Operator: https://coreos.com/blog/introducing-operators.html
+
+For more details, see the following documentation pages:
+
+- :doc:`./archive`
+- :doc:`./operator`
+- (TODO) Solution UI guidelines
+
+Once a Solution is deployed on MetalK8s, a user can deploy one or more versions
+of the Solution Operator, using either the Solution UI or the Kubernetes API,
+into separate namespaces. Using the Operator-defined ``CustomResource(s)``, the
+user can then effectively deploy the application packaged in the Solution.
+
+How is a *Solution* declared in MetalK8s?
+-----------------------------------------
+
+MetalK8s already uses a ``BootstrapConfiguration`` object, stored in
+``/etc/metalk8s/bootstrap.yaml``, to define how the cluster should be
+configured from the bootstrap node, and what versions of MetalK8s are available
+to the cluster.
+
+In the same vein, we want to use a ``SolutionsConfiguration`` object, stored in
+``/etc/metalk8s/solutions.yaml``, to declare which Solutions are available to
+the cluster, from the bootstrap node.
+
+.. todo:: Add specification in a future Reference guide
+
+Here is how it could look::
+
+    apiVersion: metalk8s.scality.com/v1alpha1
+    kind: SolutionsConfiguration
+    solutions:
+      - /solutions/storage_1.0.0.iso
+      - /solutions/storage_latest.iso
+      - /other_solutions/computing.iso
```

Comment:
This example has the advantage of showing 2 versions of the same Solution, I was going to be ask about this case. It leaves me wondering how this case would work: is it designed for parallel run (either because 2 different versions will be needed indefinitely because of independent release tracks, or because we want to support gradual migration from version `v` to version `v+1`) or only one version active at a time that need to be selected/switched by the user through API/UI?

Reply:
It's designed to run multiple versions of a Solution at the same time, using Namespaces for isolation. This allows users to try a new version in some `staging` namespace first, before upgrading their `prod` namespace.

Reply:
I improved a few sections to be more explicit about this intent, later improvements will include UML diagrams for these use-cases.